azimuth/modules/dataset_analysis/dataset_warnings.py

```diff
@@ -193,135 +194,121 @@ def get_general_warnings(
-        max_delta_std_words = int(self.config.dataset_warnings.max_delta_std_words)
-        alert_norm_mean = divergence_mean > max_delta_mean_words
-        alert_norm_std = divergence_std > max_delta_std_words
-        alert_norm = np.any([alert_norm_mean, alert_norm_std], axis=0)
+        class_index = np.arange(dm.get_num_classes(labels_only=True))
+        agg_per_split: Dict[DatasetSplitName, Dict[Agg, float]] = {
+            split: {} for split in self.available_dataset_splits
+        }
+        agg_per_label_per_split: Dict[DatasetSplitName, Dict[Agg, np.ndarray]] = {
+            split: {} for split in self.available_dataset_splits
+        }
+        hist_per_label_per_split = {}
+
+        for split in self.available_dataset_splits:
+            ds = self.get_dataset_split(split).remove_columns([])
+            df = ds.remove_columns(
+                [
+                    column
+                    for column in ds.column_names
+                    if column not in [DatasetColumn.word_count, self.config.columns.label]
+                ]
+            ).to_pandas()
+
+            agg_per_split[split][Agg.mean] = df["word_count"].mean()
+            agg_per_split[split][Agg.std] = df["word_count"].std()
+
+            # Get mean and std word count per label
+            stats_per_label = df.groupby("label").agg(["mean", "std"]).reindex(class_index)
+            for agg in [Agg.mean, Agg.std]:
+                agg_per_label_per_split[split][agg] = stats_per_label["word_count"][agg].to_numpy()
+
+            # Get word count histogram per label. Columns are word counts and rows are labels.
+            hist_per_label_per_split[split] = (
+                df.groupby(["label", "word_count"])
+                .size()
+                .unstack(fill_value=0)
+                .reindex(class_index, fill_value=0)
+            )
 
-        # Get counts, this will have N columns with count and LABEL rows.
-        train_count = (
-            train_df.groupby(["label", DatasetColumn.word_count]).size().unstack(fill_value=0)
-        )
-        eval_count = (
-            eval_df.groupby(["label", DatasetColumn.word_count]).size().unstack(fill_value=0)
+        # Compute divergence and alerts
+        divergence_per_label_per_agg = dict()
+        alert_per_label_per_agg = dict()
+        thresholds_per_agg = {
+            Agg.mean: self.config.dataset_warnings.max_delta_mean_words,
+            Agg.std: self.config.dataset_warnings.max_delta_std_words,
+        }
+        for agg in thresholds_per_agg.keys():
+            divergence_per_label_per_agg[agg] = np.abs(
+                (
+                    agg_per_label_per_split[DatasetSplitName.train][agg]
+                    - agg_per_label_per_split[DatasetSplitName.eval][agg]
+                )
+            )
+            alert_per_label_per_agg[agg] = (
+                divergence_per_label_per_agg[agg] > thresholds_per_agg[agg]
+            )
```

Comment:
Also, here, it seems there are some redundant parentheses.
```suggestion
        for agg, threshold in thresholds_per_agg.items():
            divergence_per_label_per_agg[agg] = np.abs(
                agg_per_label_per_split[DatasetSplitName.train][agg]
                - agg_per_label_per_split[DatasetSplitName.eval][agg]
            )
            alert_per_label_per_agg[agg] = divergence_per_label_per_agg[agg] > threshold
```

Reply:
Totally for the first one, but the second ones are added by the pre-commits.

Reply:
Of course, but by changing `thresholds_per_agg[agg]` to `threshold` (thanks to `.items()`), it now fits on one line.